lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -481,32 +481,32 @@ def _get_nodes_to_remove(self):
 
 
 def main():
+    """ elasticache ansible module """
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        state                 ={'required': True, 'choices': ['present', 'absent', 'rebooted']},
-        name                  ={'required': True},
-        engine                ={'required': False, 'default': 'memcached'},
-        cache_engine_version  ={'required': False},
-        node_type             ={'required': False, 'default': 'cache.m1.small'},
-        num_nodes             ={'required': False, 'default': None, 'type': 'int'},
+        state={'required': True, 'choices': ['present', 'absent', 'rebooted']},
```

Comment:
Almost all modules use `dict()` constructor here.

`required: False`, `default: None`, and `type: 'str'` are defaults and not required.

Not really a blocker, more advisory.